test/libsolidity/SemanticTest.cpp

```diff
@@ -136,49 +129,58 @@ void SemanticTest::initializeBuiltins()
 TestCase::TestResult SemanticTest::run(ostream& _stream, string const& _linePrefix, bool _formatted)
 {
 	TestResult result = TestResult::Success;
-	bool compileViaYul = m_runWithYul || m_enforceViaYul;
 
-	if (m_runWithoutYul)
+	if (m_testCaseWantsLegacyRun)
 		result = runTest(_stream, _linePrefix, _formatted, false, false);
 
-	if (compileViaYul && result == TestResult::Success)
+	if ((m_testCaseWantsYulRun || m_enforceViaYul) && result == TestResult::Success)
 		result = runTest(_stream, _linePrefix, _formatted, true, false);
 
-	if (m_runWithEwasm && result == TestResult::Success)
-		result = runTest(_stream, _linePrefix, _formatted, true, true);
-
+	if ((m_testCaseWantsEwasmRun || m_enforceCompileToEwasm) && result == TestResult::Success)
+	{
+		// TODO: Once we have full Ewasm support, we could remove try/catch here.
+		try
+		{
+			result = runTest(_stream, _linePrefix, _formatted, true, true);
+		}
+		catch (...)
+		{
+			if (!m_enforceCompileToEwasm)
+				throw;
+		}
+	}
 	return result;
 }
 
 TestCase::TestResult SemanticTest::runTest(
 	ostream& _stream,
 	string const& _linePrefix,
 	bool _formatted,
-	bool _compileViaYul,
-	bool _compileToEwasm
-)
+	bool _isYulRun,
+	bool _isEwasmRun)
 {
 	bool success = true;
 	m_gasCostFailure = false;
 
-	if (_compileViaYul && _compileToEwasm)
+	if (_isYulRun && _isEwasmRun)
 		selectVM(evmc_capabilities::EVMC_CAPABILITY_EWASM);
```

Comment:
Doesn't ewasm imply Yul?
```suggestion
	if (_isEwasmRun)
	{
		soltestAssert(_isYulRun, "");
		selectVM(evmc_capabilities::EVMC_CAPABILITY_EWASM);
	}
```